services/stickers/transactions.go

```diff
@@ -65,23 +71,99 @@ func (api *API) Buy(ctx context.Context, chainID uint64, txArgs transactions.Sen
 		return "", err
 	}
 
-	err = api.AddPending(chainID, packID)
+	api.txSignDetails = &txSigningDetails{
+		txType:        transactions.BuyStickerPack,
+		chainID:       chainID,
+		from:          txOpts.From,
+		txBeingSigned: tx,
+		packID:        packID,
+	}
+
+	signer := ethTypes.NewLondonSigner(new(big.Int).SetUint64(api.txSignDetails.chainID))
+	return signer.Hash(api.txSignDetails.txBeingSigned), nil
+}
+
+// Signs a transaction using appropriate local keystore file that is decrypted with provided password and returns the signature in hex format.
+// The tx that will be signed is the one that was prepared by the last call to PrepareTxForBuyingStickers method.
+func (api *API) SingPreparedTx(password string) (interface{}, error) {
```

Comment:
The code is the same as in ens/api.go. Can you move it to eg. services/utils/utils.go? There is a GetSigner function which you can use also.

Reply:
> There is a GetSigner function which you can use also.

Well, it looks similar, but it doesn't do exactly the same thing as `GetSigner` does. 
- `GetSigner`:
  - makes a tx hash for signing
  - signs tx hash building a signature
  - sends a tx using prepared signature and returns tx of sent tx
- `SingPreparedTx`:
  - signs prepared tx and returns signature
  
> Can you move it to eg. services/utils/utils.go?

Yes, those 2 are very similar, basically they do the same thing, but over different tx. One signs a tx prepared in the context of ens context, other signs a tx prepared in the stickers context. But yes, I can move those few lines, the common part, to the utils.go file, I agree.